app/javascript/packages/document-capture/components/address-search.tsx

```diff
@@ -0,0 +1,47 @@
+import { TextInput, Button } from '@18f/identity-components';
+import { request } from '@18f/identity-request';
+import { useState, useCallback } from 'react';
+
+interface Location {
+  street_address: string;
+  city: string;
+  state: string;
+  zip_code: string;
+  address: string;
+}
+
+interface AddressSearchProps {
+  onAddressFound?: (location: Location) => void;
+}
+
+export const ADDRESS_SEARCH_URL = '/api/addresses';
+
+function AddressSearch({ onAddressFound = () => {} }: AddressSearchProps) {
+  const [unvalidatedAddressInput, setUnvalidatedAddressInput] = useState('');
+  const [addressQuery, setAddressQuery] = useState({} as Location);
+  const handleAddressSearch = useCallback(async () => {
+    const addressCandidates = await request(ADDRESS_SEARCH_URL, {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      json: { address: unvalidatedAddressInput },
+    });
+
+    const [bestMatchedAddress] = addressCandidates;
```

Comment:
is this being reassigned in order to make addresscandidates an array?

Reply:
so this grabs the first address from addressCandidates?

Reply:
No strong preference to me, but to address potential readability concerns with destructuring, maybe we just pick off the first explicitly?

```suggestion
    const bestMatchedAddress = addressCandidates[0];
```